### PR TITLE
Basic QR redirection on scan

### PR DESCRIPTION
**Pull request by Valentin889**

### CODE REVIEW

**Comment by skyrag** [comment]
qr scanning is working well, nice addition to the app

File: app/src/main/java/com/android/partagix/ui/App.kt [hunk 1]
```diff
@@ -78,14 +84,61 @@ class App(
...
     fusedLocationClient = LocationServices.getFusedLocationProviderClient(activity)
     ComposeNavigationSetup()
 
-    navigationActions.navigateTo(Route.BOOT)
+    if (idItem != null && FirebaseAuth.getInstance().currentUser != null) {
+      onQrScanned(idItem, FirebaseAuth.getInstance().currentUser!!.uid)
+    } else {
+      navigationActions.navigateTo(Route.BOOT)
+    }
+    /*
```
Comment (skyrag) [inline app/src/main/java/com/android/partagix/ui/App.kt:100]: open and end comment without anything in it

File: app/src/main/java/com/android/partagix/model/StartOrEndLoanViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    val loan = _uiState.value.loan
+    val newLoan =
+        loan.copy(
+            state = LoanState.ONGOING,
+            startDate = Date(),
+            endDate = Date(),
+        )
+    db.setLoan(newLoan)
+    // db.setItem(item.copy(state = ItemState.BORROWED))
+    // db.updateUser(borrower.copy(loanId = newLoan.id))
```
Comment (skyrag) [inline app/src/main/java/com/android/partagix/model/StartOrEndLoanViewModel.kt:35]: are those commented functions usefull or cant we just erase those lines?

Comment (Valentin889) [reply app/src/main/java/com/android/partagix/model/StartOrEndLoanViewModel.kt]: Fixed

File: app/src/main/java/com/android/partagix/ui/App.kt (hunk 1 above)
Comment (Valentin889) [reply app/src/main/java/com/android/partagix/ui/App.kt]: Fixed

File: app/src/main/java/com/android/partagix/model/StartOrEndLoanViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    _uiState.value = new
+  }
+
+  fun onStart() {
+    val loan = _uiState.value.loan
+    val newLoan =
+        loan.copy(
+            state = LoanState.ONGOING,
+            startDate = Date(),
+            endDate = Date(),
```
Comment (skyrag) [inline app/src/main/java/com/android/partagix/model/StartOrEndLoanViewModel.kt:31]: i saw that when you begin your loan you change the end_date of the loan to the actual date of the beginning of the loan which is problematic. If I'm not mistaken the error is on this line

**Review by skyrag — CHANGES_REQUESTED** [review]
the end_date change at the same time as the start_date

**Review by skyrag — APPROVED** [review]
Nice Pr and code in general, and the few little problems where resolved

**Pull request merged**